File: Test/Test/ViewController.m
//
//  ViewController.m
//  Test
//
//  Created by Xenosxiny on 16/9/7.
//  Copyright © 2016年 Xenosxiny. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()
@property(nonatomic,strong) UIButton *button ;
@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    CGRect frame = self.button.frame ;
    frame.origin.x = 120 ;
    [self.view addSubview:self.button] ;
}


- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
    
}



-(UIButton *)button{
    _button = [[UIButton alloc]init] ;
    _button.frame = CGRectMake(100, 100, 100, 100) ;
    _button.backgroundColor = [UIColor redColor] ;
    _button.titleLabel.text = @"chile" ;
    return _button ;
    
}
//-(void)clickMe:(UIButton*)sender{
//    NSLog(@"1") ;
//}
-(void)clickMe:(UIButton*)sender AndInt:(int)a{
    NSLog(@"1") ;
}


@end

File: Emoji_StackView/Emoji_StackView/ViewController.swift
//
//  ViewController.swift
//  Emoji_StackView
//
//  Created by Xenosxiny on 16/9/2.
//  Copyright © 2016年 Xenosxiny. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    
    @IBOutlet var emojiCollection: [UIButton]!{
        didSet{
            emojiCollection.forEach {
                $0.hidden = true
            }
        }
    }
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
    }

    @IBAction func displayEmoji(sender: UIButton) {
        UIView.animateWithDuration(1) {
            self.emojiCollection.forEach({
                $0.hidden = !$0.hidden
            })
        }
    }
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }


}

